src/main/java/org/betonquest/betonquest/commands/QuestCommand.java

```diff
@@ -1813,6 +1830,146 @@ private Optional<List<String>> completeDownload(final String... args) {
         };
     }
 
+    /**
+     * Variables stuff.
+     */
+    @SuppressWarnings("PMD.NcssCount")
+    private void handleVariables(final CommandSender sender, final String... args) {
+        final Profile profile = getTargetProfile(sender, args);
+        if (profile == null) {
+            return;
+        }
+
+        final boolean isOnline = profile.getOnlineProfile().isPresent();
+        if (!isOnline) {
+            log.debug("Can't access variable data on offline player");
+            sendMessage(sender, "offline_invalid");
+            return;
+        }
+
+        if (args.length == 2) {
+            log.debug("Missing objective instruction string");
+            sendMessage(sender, "specify_objective");
+            return;
+        }
+
+        if (args.length == 3) {
+            log.debug("Missing argument");
+            sendMessage(sender, "arguments");
+            return;
+        }
```

Comment:
Yes, the `if` followed by a `switch` was strange, but I wouldn't remove the shortcut entirely because of that. Other commands have the same logic (just implemented in that weird way), but we could do it here like this:

```suggestion
        final String subCommand;
        if (args.length == 3) {
            subCommand = "list";
        } else {
            subCommand = args[3].toLowerCase(Locale.ROOT);
        }
```